### PR TITLE
Fixes for version deletion.

### PR DESCRIPTION
**Pull request by paulgoleary**
Pulls out Marc's fix for deleting most recent version hash correctly. Previous code was not 'pulling' and older version in the current `objectHash` variable if the most recent version was deleted.

Added to existing tests in `elv-master` which previously did not cover this case.

### CODE REVIEW

File: editable.sol [hunk 1]
```diff
@@ -110,14 +112,42 @@ contract Editable is Ownable {
...
+                        mostRecent = x;
+                        latestStamp = versionTimestamp[x];
+                    }
+                }
+                // assign most recent version as object version and delete from versions array
+                objectHash = versionHashes[mostRecent];
+                objectTimestamp = latestStamp;
+                removeVersionIdx(mostRecent);
+            }
             emit VersionDelete(contentSpace, _versionHash, 0);
```
Comment (elv-preethi) [inline editable.sol]: should the event call ` emit VersionDelete(contentSpace, _versionHash, 0);` be added to `function removeVersionIdx(uint idx){}` to display right index?

Comment (paulgoleary) [reply editable.sol]: I think what's there is correct right now. When the hash to be removed is the 'current' value then it's logically ordinal zero.

Actually, with the way the implementation has changed, emitting the index doesn't have that much meaning anymore. Perhaps we should emit the timestamp.

Comment (elv-preethi) [reply editable.sol]: Got it. Also the function is called at line 159, should we emit the event at that line as well.

**Comment by paulgoleary** [comment]
Closing - subsumed by `authv3changes`  - #52.

**Pull request closed without merging**